pkg/daemon/ceph/client/pool.go

```diff
@@ -471,21 +471,21 @@ func createReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterI
 
 	logger.Infof("reconciling replicated pool %s succeeded", pool.Name)
 
-	if checkFailureDomain {
-		if err = ensureFailureDomain(context, clusterInfo, clusterSpec, pool); err != nil {
+	if checkFailureDomain || pool.PoolSpec.DeviceClass != "" {
+		if err = updatePoolCrushRule(context, clusterInfo, clusterSpec, pool); err != nil {
 			return nil
 		}
 	}
 	return nil
 }
 
-func ensureFailureDomain(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, pool cephv1.NamedPoolSpec) error {
-	if pool.FailureDomain == "" {
-		logger.Debugf("skipping check for failure domain on pool %q as it is not specified", pool.Name)
+func updatePoolCrushRule(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, pool cephv1.NamedPoolSpec) error {
+	if pool.FailureDomain == "" && pool.DeviceClass == "" {
+		logger.Debugf("skipping check for failure domain %q and deviceClass on pool %q as it is not specified", pool.Name, pool.DeviceClass)
```

Comment:
deviceClass is always empty here, no need to format it in the message
```suggestion
		logger.Debugf("skipping check for failure domain and deviceClass on pool %q as it is not specified", pool.Name)
```

Reply:
output with testing 
```
 skipping check for failure domain and deviceClass "replicapool" on pool as it is not specified
```